pkg/apis/apps.kubermatic/v1/types.go

```diff
@@ -23,4 +23,7 @@ const (
 	// ApplicationInstallationCleanupFinalizer indicates that application installed on user-cluster need cleanup
 	// ie uninstall the application, remove  namespace where application were installed ...
 	ApplicationInstallationCleanupFinalizer = "kubermatic.k8c.io/cleanup-application-installation"
+
+	// ApplicationSecretCleanupFinalizer indicates that secret synced from kubermatic namespace to cluster namespace need cleanup.
+	ApplicationSecretCleanupFinalizer = "kubermatic.k8c.io/cleanup-application-secret"
```

Comment:
Does this need to be a global finalizer or could this simply be a private const in a controller package?

Reply:
can do in the controller